resources/key.rb

```diff
@@ -0,0 +1,103 @@
+
+@section Examples
+
+    rundeck_server_key 'mykey' do
+      type 'private'
+      content '---START.....'
+      endpoint 'ttp://127.0.0.1:4440'
+      api_token 'myticket''
+      action :create
+    end
+#>
+=end
+
+# <> @property path to key
+property :key,    kind_of: String, name_property:  true
+# <> @property tyoe of key. Can be 'private', 'public'
+property :type,      kind_of: String, required: true
+# <> @property key data
+property :content,    kind_of: String, required: false
+# <> @property endpoint
+property :endpoint,  kind_of: String, default: 'http://127.0.0.1:4440'
+# <> @property api_token Token used to interact with the api. See rundeck documentation to generate a token.
+property :api_token, kind_of: String, required: true
+
+action :create do
+  require 'rundeck'
+
+  client = Rundeck.client(endpoint: @new_resource.endpoint, api_token: @new_resource.api_token)
+  if key_exists?(client, @new_resource.key)
+    Chef::Log.warn("Storage key #{@new_resource.key} already exists, updating...")
+    converge_by "Update storage key #{@new_resource.key}" do
+      begin
+        case @new_resource.type
+        when 'private'
+          response = client.update_private_key(@new_resource.key, @new_resource.content)
+        when 'public'
+          response = client.update_public_key(@new_resource.key, @new_resource.content)
+        else
+         fail 'Supported types: [private, public]'
+        end
+      rescue Rundeck::Error::Forbidden
+        fail "Forbidden access to #{client.endpoint} with api token '#{client.api_token}'"
+      end
+      Chef::Log.debug('Result: ' + response.inspect)
+    end
+
+  else
+    converge_by "Create storage key #{@new_resource.key}" do
+      begin
+        case @new_resource.type
+        when 'private'
+          response = client.create_private_key(@new_resource.key, @new_resource.content)
+        when 'public'
+          response = client.create_public_key(@new_resource.key, @new_resource.content)
+        else
+         fail 'Supported types: [private, public]'
+        end
+      rescue Rundeck::Error::Forbidden
+        fail "Forbidden access to #{client.endpoint} with api token '#{client.api_token}'"
+      end
+      Chef::Log.debug('Result: ' + response.inspect)
+    end
+  end
+end
+
+action :delete do
+  require 'rundeck'
+
+  client = Rundeck.client(endpoint: @new_resource.endpoint, api_token: @new_resource.api_token)
+  if key_exists?(client, @new_resource.key)
+    Chef::Log.warn("Storage key #{@new_resource.key} already exists, updating...")
+    converge_by "Delete storage key #{@new_resource.key}" do
+      begin
+        response = client.delete_key(@new_resource.key)
+      rescue Rundeck::Error::Forbidden
+        fail "Forbidden access to #{client.endpoint} with api token '#{client.api_token}'"
+      end
+      Chef::Log.debug('Result: ' + response.inspect)
+    end
+
+  else
+    Chef::Log.warn("Storage key #{@new_resource.key} doesn't exists!")
```

Comment:
this should not be a warning since this is a "normal" outcome (which will happen on all chef-client run except the one actually deleting the resource)

Reply:
OK, removed